### PR TITLE
remove DataTypeInfo and use LogicalType and column names

### LINKED ISSUE #2423: Remove DataTypeInfo from codebase
```
struct DataTypeInfo {
public:
    DataTypeInfo(common::LogicalTypeID typeID, std::string name)
        : typeID{typeID}, name{std::move(name)}, numValuesPerList{0} {}

    common::LogicalTypeID typeID;
    std::string name;
    std::vector<std::unique_ptr<DataTypeInfo>> childrenTypesInfo;
    // Used by fixedList only.
    uint64_t numValuesPerList;

    static std::unique_ptr<DataTypeInfo> getInfoForDataType(
        const common::LogicalType& type, const std::string& name);
};
```
We should remove this class and use LogicalType, columnNames directly.

### PR DESCRIPTION
**Pull request by russell-liu**
fixes #2423

### CODE REVIEW

File: src/common/arrow/arrow_row_batch.cpp [hunk 1]
```diff
@@ -47,141 +46,139 @@ void ArrowRowBatch::templateInitializeVector<LogicalTypeID::STRING>(
...
         vector->childData.push_back(std::move(childVector));
     }
 }
 
 template<>
 void ArrowRowBatch::templateInitializeVector<LogicalTypeID::INTERNAL_ID>(
-    ArrowVector* vector, const main::DataTypeInfo& typeInfo, std::int64_t capacity) {
-    initializeStructVector(vector, typeInfo, capacity);
+    ArrowVector* vector, const LogicalType& type, std::int64_t capacity) {
+    initializeStructVector(vector, type, capacity);
```
Comment (andyfengHKU) [inline src/common/arrow/arrow_row_batch.cpp:86]: Add test to cover all uncovered lines.

File: src/include/common/arrow/arrow_row_batch.h [hunk 2]
```diff
@@ -44,46 +44,43 @@ struct ArrowVector {
...
     static void templateCopyNullValue(ArrowVector* vector, std::int64_t pos);
     template<LogicalTypeID DT>
-    static ArrowArray* templateCreateArray(ArrowVector& vector, const main::DataTypeInfo& typeInfo);
+    static ArrowArray* templateCreateArray(ArrowVector& vector, const LogicalType& type);
 
     ArrowArray toArray();
 
 private:
-    std::vector<std::unique_ptr<main::DataTypeInfo>> typesInfo;
+    std::vector<LogicalType> types;
```
Comment (andyfengHKU) [inline src/include/common/arrow/arrow_row_batch.h:83]: use `common::logical_types_t`

**Review by andyfengHKU — APPROVED** [review]

**Comment by russell-liu** [comment]
After attempting to add more Arrow tests, I've discovered a problem with this removal of DataTypeInfo that I do not know how to resolve. Details:

I tried to return a node, rel, or internal ID in the Arrow tests, and this led to segmentation faults. After further investigation, I believe this is because the following logic in the old implementation using DataTypeInfo is not preserved:
https://github.com/kuzudb/kuzu/blob/master/src/main/query_result.cpp#L24
https://github.com/kuzudb/kuzu/blob/master/src/main/query_result.cpp#L102
https://github.com/kuzudb/kuzu/blob/master/src/main/query_result.cpp#L115
That is, when DataTypeInfo used to be created for internal IDs, nodes, or rels, some additional childrenTypesInfo would be created; when we switch to using DataType only, these fields are missing.

I tried to modify arrow_row_batch.cpp (including templateInitializeVector and templateCopyNonNullValue for internal IDs, nodes, and rels) to replicate this logic, but I didn't really know what I was doing and I ended up with a segmentation fault occurring during std::memcpy that I didn't know what to do about. My approach was probably incorrect and messy I won't push the code here.

**Comment by mxwli** [comment]
Should close #3224 as well

File: Makefile [hunk 3]
```diff
@@ -111,7 +111,7 @@ nodejs:
 	$(call run-cmake-release, -DBUILD_NODEJS=TRUE)
 
 python:
-	$(call run-cmake-release, -DBUILD_PYTHON=TRUE)
+	$(call run-cmake-debug, -DBUILD_PYTHON=TRUE)
```
Comment (andyfengHKU) [inline Makefile:114]: Shouldn't we make release for python test by default?

File: tools/python_api/src_cpp/py_query_result.cpp [hunk 4]
```diff
@@ -325,8 +325,8 @@ py::object PyQueryResult::getAsDF() {
     return QueryResultConverter(queryResult.get()).toDF();
 }
 
-bool PyQueryResult::getNextArrowChunk(const std::vector<std::unique_ptr<DataTypeInfo>>& typesInfo,
-    py::list& batches, std::int64_t chunkSize) {
+bool PyQueryResult::getNextArrowChunk(const std::vector<kuzu::common::LogicalType>& types,
```
Comment (andyfengHKU) [inline tools/python_api/src_cpp/py_query_result.cpp:328]: remove `kuzu::common::` check all files in this PR.

File: src/include/common/arrow/arrow_row_batch.h [hunk 5]
```diff
@@ -44,46 +44,46 @@ struct ArrowVector {
...
     static void templateCopyNullValue(ArrowVector* vector, std::int64_t pos);
     template<LogicalTypeID DT>
-    static ArrowArray* templateCreateArray(ArrowVector& vector, const main::DataTypeInfo& typeInfo);
+    static ArrowArray* templateCreateArray(ArrowVector& vector, const LogicalType& type);
 
     ArrowArray toArray();
 
 private:
-    std::vector<std::unique_ptr<main::DataTypeInfo>> typesInfo;
+    common::logical_types_t types;
```
Comment (andyfengHKU) [inline src/include/common/arrow/arrow_row_batch.h:86]: use `std::vector<common::LogicalType>`. I'm planning to remove `common::logical_types_t`

File: src/common/arrow/arrow_type.cpp [hunk 6]
```diff
@@ -105,7 +105,7 @@ LogicalType ArrowConverter::fromArrowSchema(const ArrowSchema* schema) {
         case 'w':
             throw RuntimeException("Fixed list is currently WIP.");
             // TODO Manh: Array Binding
-            // return *LogicalType::FIXED_LIST(
```
Comment (andyfengHKU) [inline src/common/arrow/arrow_type.cpp:108]: Open an issue for `Manh` to revisit this.

**Review by andyfengHKU — APPROVED** [review]
Make sure we add corresponding test cases for coverage.

File: Makefile (hunk 3 above)
Comment (mxwli) [reply Makefile:114]: Yes. I need to stop leaving these changes in.

**Pull request merged**